### ISSUE TITLE
Segfault when running anvi-profile

### ISSUE THREAD

**Issue by maureenbug** [merenlab/anvio#724]
Hi!

I am trying to run anvi'o, and I keep running into a segfault when running the anvi-profile command (see command and output below). I don't think my deflines are a problem (e.g. NODE_1_length_127577_cov_292_479718).

I am running on a Linux server, and installed anvi'o via bioconda.

Thanks in advance!
Maureen

```
:~> anvi-self-test --version
Anvi'o version ...............................: 2.1.0
Profile DB version ...........................: 17
Contigs DB version ...........................: 8
Pan DB version ...............................: 4
Samples information DB version ...............: 2
Genome data storage version ..................: 1
Auxiliary data storage version ...............: 3
Anvi'server users data storage version .......: 1


:~> anvi-profile -i /projectb/scratch/mberg/1187472_MAPPING_fixed/Sample_1187472.bam -c anvio_1187472_fixed_contigs.db --output-dir /projectb/scratch/mberg/anvio_files/1187472_anvio/anvio_profile/ --sample-name a1187472
anvio ..............................: 2.1.0
profiler_version ...................: 17
sample_id ..........................: a1187472
profile_db .........................: /projectb/scratch/mberg/anvio_files/1187472_anvio/anvio_profile/PROFILE.db
contigs_db .........................: True
contigs_db_hash ....................: 9e02e2b1
cmd_line ...........................: /global/homes/m/mberg/.conda/envs/mgpipe/bin/anvi-profile -i /projectb/scratch/mberg/1187472_MAPPING_fixed/Sample_1187472.bam -c anvio_1187472_fixed_contigs.db --output-dir /projectb/scratch/mberg/anvio_files/1187472_anvio/anvio_profile/ --sample-name a1187472
merged .............................: False
blank ..............................: False
split_length .......................: 20,000
min_contig_length ..................: 2,500
min_mean_coverage ..................: 0
clustering_performed ...............: False
min_coverage_for_variability .......: 10
skip_SNV_profiling .................: False
profile_AA_frequencies .............: False
report_variability_full ............: False
gene_coverages_computed ............: 1
input_bam ..........................: /projectb/scratch/mberg/1187472_MAPPING_fixed/Sample_1187472.bam                                                                                                                                                                        
output_dir .........................: /projectb/scratch/mberg/anvio_files/1187472_anvio/anvio_profile
total_reads_mapped .................: 29,973,982
num_contigs ........................: 3,121
num_contigs_after_M ................: 1,429
num_splits .........................: 1,479
total_length .......................: 10,826,783
[31 Jan 18 15:23:42 Profiling "NODE_1_length_127577_cov_292_479718" (1 of 1429) (127,577 nts)] Coverage (split: 6 of 6)                                                                                                                                                      Segmentation fault
```

**Comment by ozcan** [comment]
Hi @maureenbug,

I am suspecting that this segmentation fault is coming from pysam. Can you try the commands below to see if it is actually coming from pysam?

```
python -c "import pysam; pysam.Index('/projectb/scratch/mberg/1187472_MAPPING_fixed/Sample_1187472.bam')"
```

```
python -c "import pysam; p = pysam.Samfile('/projectb/scratch/mberg/1187472_MAPPING_fixed/Sample_1187472.bam','rb'); print(p.pileup('NODE_1_length_127577_cov_292_479718', 0, 100))"
```

It should print something like `<pysam.libcalignmentfile.IteratorColumnRegion object at 0x110240708>` or it will give a segmentation fault

All the best,
Ozcan

**Comment by meren** [comment]
Also, it would have been excellent if you were to upgrade your anvi'o to the latest version, @maureenbug. `v2.1.0` is 14 years old now :(

**Comment by maureenbug** [comment]
Here's the output:

mberg@dint04:~> python -c "import pysam; pysam.Index('/projectb/scratch/mberg/1187472_MAPPING_fixed/Sample_1187472.bam')"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: 'module' object has no attribute 'Index'


mberg@dint04:~> python -c "import pysam; p = pysam.Samfile('/projectb/scratch/mberg/1187472_MAPPING_fixed/Sample_1187472.bam','rb'); print(p.pileup('NODE_1_length_127577_cov_292_479718', 0, 100))"
<pysam.calignmentfile.IteratorColumnRegion object at 0x2ab959ac6d98>

Really?? 14 years old? I'm surprised since I just installed it last week, but I'll update it since bioconda obviously didn't install the latest version!

**Comment by meren** [comment]
Are you sure you installed it by specifying Python 3?

```
conda create -n anvio3 -c bioconda -c conda-forge python=3.5.4 gsl anvio
```

This should install anvi'o `v3`, which may resolve the issue you run into with the profiling step automatically (since `v2.1.0` we did update the way anvi'o profiles stuff dramatically, which was about 42 years ago as you now know).


Best,

**Comment by maureenbug** [comment]
I did not install it specifying Python 3, but I have the latest version now and I'm no longer running into the segfault issue. Thanks!

I did run into an error about internal stop codons (even though I used `gff_parser.py`), but I put in the `--ignore-internal-stop-codons` flag in the `anvi-gen-contigs-database` command for now. This could be an issue with something other than anvi'o, though.

PS I installed anvi'o originally using this, which installed that old version:
`conda install anvio`
